### PR TITLE
games-section-list of games

### PR DESCRIPTION
**Pull request by Bestrafer**

### CODE REVIEW

File: src/games/components/games.js [hunk 1]
```diff
@@ -2,30 +2,56 @@ import React from 'react';
 import * as gameActions from '../actions/games';
```
Comment (michalinasowa) [inline src/games/components/games.js]: It's a good practice to have imports from libraries first and then your own files, e.g.:
```
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { Table } from 'antd';

import * as gameActions from '../actions/games';
```

File: src/games/components/games.js [hunk 2]
```diff
@@ -2,30 +2,56 @@ import React from 'react';
...
+const columns = [{
+    title: 'Title',
+    dataIndex: 'title',
+    key: 'title',
+}, {
+    title: 'Owner',
+    dataIndex: 'owner',
+    key: 'owner',
+}, {
+    title: 'minPlayers',
```
Comment (michalinasowa) [inline src/games/components/games.js]: Is it userfriendly table header? :smile:

File: src/games/components/games.js [hunk 3]
```diff
@@ -2,30 +2,56 @@ import React from 'react';
...
-    }
-
-    handleClick() {
         this.props.actions.getGames();
     }
 
-    showGames() {
-        console.log(this.props.games);
+    pushGamesToTable() {
+        let gameList = this.props.games.games;
```
Comment (michalinasowa) [inline src/games/components/games.js]: `games.games` is strange... Maybe change `games` to entities in your reducer and then in `mapStateToProps` return games.entities as games ( `games: state.games.enities`).

Comment (michalinasowa) [inline src/games/components/games.js]: and it's better to use `const` not `let` if you can :) You shouldn't be able to modify variables.

File: src/games/components/games.js [hunk 4]
```diff
@@ -2,30 +2,56 @@ import React from 'react';
...
+
+        for (var key in gameList) {
+            dataSource = dataSource.concat([{
+                key: key,
```
Comment (michalinasowa) [inline src/games/components/games.js]: In ES6 you can simply write `key` not `key: key`

File: src/games/components/games.js [hunk 5]
```diff
@@ -2,30 +2,56 @@ import React from 'react';
...
+    title: 'minPlayers',
+    dataIndex: 'minPlayers',
+    key: 'minPlayers',
+}, {
+    title: 'maxPlayers',
+    dataIndex: 'maxPlayers',
+    key: 'maxPlayers',
+}];
+
+var dataSource = [];
```
Comment (michalinasowa) [inline src/games/components/games.js]: global variable? :fearful:

File: src/games/components/games.js [hunk 6]
```diff
@@ -2,30 +2,56 @@ import React from 'react';
...
     render() {
+        this.pushGamesToTable();
+
         return (
             <div>
-                <div>Hania Games!</div>
-
-                <button onClick={this.handleClick}>GET GAMES</button>
-                <button onClick={this.showGames}>SHOW GAMES</button>
+                <h1>GAMES:</h1>
```
Comment (michalinasowa) [inline src/games/components/games.js]: Don't use capital letters in code, use `texttransform: uppercase;` in CSS (or maybe do it when you'll have sass configurated)

Now you can change it to "games" and then add uppercase in styles :)

File: src/games/reducers/games.js [hunk 7]
```diff
@@ -6,7 +6,6 @@ export function gamesReducer(state = initialState, action) {
     switch (action.type) {
         case 'GAMES_FETCHED' :
         {
-            console.log(action.games);
             return Object.assign({}, state, {games: action.games});
```
Comment (michalinasowa) [inline src/games/reducers/games.js]: `{ key: value }`

it's much easier to read :)

File: src/games/reducers/games.js [hunk 8]
```diff
@@ -6,7 +6,6 @@ export function gamesReducer(state = initialState, action) {
     switch (action.type) {
         case 'GAMES_FETCHED' :
```
Comment (michalinasowa) [inline src/games/reducers/games.js]: you don't need space before `:`

File: src/games/components/games.js [hunk 9]
```diff
@@ -2,30 +2,56 @@ import React from 'react';
...
+    pushGamesToTable() {
+        let gameList = this.props.games.games;
+
+        for (var key in gameList) {
+            dataSource = dataSource.concat([{
+                key: key,
+                title: gameList[key].title,
+                owner: gameList[key].owner.name,
+                minPlayers: gameList[key].minNumber,
+                maxPlayers: gameList[key].maxNumber
```
Comment (michalinasowa) [inline src/games/components/games.js]: It's a good practice to add `,` at the end of every line in object. It'll be easier to merge code in the future :)

File: src/games/components/games.js [hunk 10]
```diff
@@ -2,30 +2,56 @@ import React from 'react';
...
+    key: 'minPlayers',
+}, {
+    title: 'maxPlayers',
+    dataIndex: 'maxPlayers',
+    key: 'maxPlayers',
+}];
+
+var dataSource = [];
 
 class GamesContainer extends React.Component {
```
Comment (michalinasowa) [inline src/games/components/games.js]: try to rewrite this component to stateless component :)

File: src/games/components/games.js (hunk 3 above)
Comment (michalinasowa) [inline src/games/components/games.js]: or rewrite this function:
- put this.props.games.entities ( :smile: ) as parameter to the function

It's the first step to make your component stateless (you don't need `this` in this function anymore)

File: src/games/components/games.js (hunk 1 above)
Comment (Bestrafer) [inline src/games/components/games.js]: :+1:

File: src/games/components/games.js (hunk 2 above)
Comment (Bestrafer) [inline src/games/components/games.js]: :+1:

File: src/games/components/games.js (hunk 5 above)
Comment (Bestrafer) [inline src/games/components/games.js]: What do you suggest instead?

File: src/games/components/games.js (hunk 10 above)
Comment (Bestrafer) [inline src/games/components/games.js]: what is stateless component and what are advantages to do it this way?

File: src/games/components/games.js (hunk 4 above)
Comment (Bestrafer) [inline src/games/components/games.js]: :+1:

File: src/games/components/games.js (hunk 9 above)
Comment (Bestrafer) [inline src/games/components/games.js]: :+1:

File: src/games/reducers/games.js (hunk 8 above)
Comment (Bestrafer) [inline src/games/reducers/games.js]: :+1:

File: src/games/reducers/games.js (hunk 7 above)
Comment (Bestrafer) [inline src/games/reducers/games.js]: :+1:

File: src/games/components/games.js (hunk 6 above)
Comment (Bestrafer) [inline src/games/components/games.js]: :+1:

File: src/games/components/games.js (hunk 10 above)
Comment (michalinasowa) [inline src/games/components/games.js]: I mean functional component and the best explanation for it is this article:
https://camjackson.net/post/9-things-every-reactjs-beginner-should-know#write-functional-components

Comment (michalinasowa) [inline src/games/components/games.js]: and read about stateless components too :wink:

File: src/games/components/games.js (hunk 5 above)
Comment (michalinasowa) [inline src/games/components/games.js]: You can for example remove it and remove function pushGamesToTable :)
Things that you are doing in "pushGamesToTable" should be done in reducer.

**Pull request merged**